syntax = "proto3";

import "common.proto";

package motify.music;

service ArtistService{
  rpc GetArtist(Id) returns (Artist) {}
  rpc GetArtists(common.Ids) returns (Artists) {}
  rpc CreateArtist(Artist) returns (Artist) {}
}

message Artists {
  repeated Artist artists = 1;
}

message Id {
  string id = 1;
}

message Artist {
  optional string id = 1;
  string name = 2;
  string user = 3;
  uint64 no_plays = 4;
}